pkg/util/span/frontier.go

```diff
@@ -439,3 +439,19 @@ func (f *Frontier) String() string {
 	})
 	return buf.String()
 }
+
+// FrontierForSpan returns the lowest timestamp in the frontier within the given
+// subspans.  If the subspans are entirely outside the Frontier's tracked span
+// an empty timestamp is returned.
```

Comment:
I don't know how I feel about including this change here.
It's a utility function currently used just for stream ingestion; perhaps it can just stay
a utility function there.

Reply:
Moved it to the ingestion processor which is the only place its used now :+1: